internal/exec/util/unit.go

```diff
@@ -15,42 +15,55 @@
 package util
 
 import (
-	"bytes"
 	"fmt"
-	"io/ioutil"
+	"net/url"
 	"os"
 	"path/filepath"
 
 	"github.com/coreos/ignition/config/types"
+
+	"github.com/vincent-petithory/dataurl"
 )
 
 const (
 	presetPath               string      = "/etc/systemd/system-preset/20-ignition.preset"
 	DefaultPresetPermissions os.FileMode = 0644
 )
 
-func FileFromSystemdUnit(unit types.Unit) *File {
-	return &File{
-		Path:       filepath.Join(SystemdUnitsPath(), string(unit.Name)),
-		ReadCloser: ioutil.NopCloser(bytes.NewReader([]byte(unit.Contents))),
-		Mode:       DefaultFilePermissions,
+func FileFromSystemdUnit(unit types.Unit) (*File, error) {
+	u, err := url.Parse(dataurl.EncodeBytes([]byte(unit.Contents)))
```

Comment:
This doesn't appear to be used outside of this function. Am I missing something?